### ISSUE TITLE
performance enhancement with onert as a single .a file instead of .so ?

### ISSUE THREAD

**Issue by chunseoklee** [Samsung/ONE#3816]
A colleague told me that there is a performance boost with tensorflow-lite.a file over tensorflow-lite.so. 

Is this applicable to onert ?

**Comment by seanshpark** [comment]
maybe with `Link Time Optimization` ?

**Comment by ragmani** [comment]
What is purpose? If just benchmark, it seems good for a little performance boost. But if for applying it to the framework, it doesn't looks good.

This is simple explanation for "Shared (dynamic) Libraries vs. Static Libraries"
https://medium.com/@nickolasteixeira/shared-dynamic-libraries-vs-static-libraries-differences-in-performance-2716f5b3c826#:~:text=Dynamic%20linking%20defers%20much%20of,are%20executed%20in%20the%20system.

**Comment by chunseoklee** [comment]
> What is purpose? If just benchmark, it seems good for a little performance boost. But if for applying it to the framework, it doesn't looks good.

In case of tensorflow-lite, there is huge enhancement by change. Thus, we may need to consider proper scenario.

**Comment by glistening** [comment]
@chunseoklee Yes, many projects that I've involved provides the monolithic (all-in-ONE) build for performance or binary size. Optimization using toolchain option is always in my optimization item list. I am thinking to do in last step.

> Is this applicable to onert ?

I believe so if we limit configuration and backend (use cpu backend only and assume linear executor) though it requires some work.
